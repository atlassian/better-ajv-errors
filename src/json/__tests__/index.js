import { readFileSync } from 'fs';
import { getFixturePath } from 'jest-fixtures';
import { parse, getMetaFromPath } from '../';

async function loadScenario(n) {
  const fixturePath = await getFixturePath(__dirname, `scenario-${n}.json`);
  return readFileSync(fixturePath, 'utf8');
}

describe('JSON', () => {
  it('can work on simple JSON', async () => {
    const rawJson = await loadScenario(1);
    expect(parse(rawJson)).toMatchSnapshot();
    expect(getMetaFromPath(rawJson, '/foo')).toMatchSnapshot();
  });

  it('can work on JSON with a key named value', async () => {
    const rawJson = await loadScenario(2);
    expect(parse(rawJson)).toMatchSnapshot();
    expect(getMetaFromPath(rawJson, '/value')).toMatchSnapshot();
  });

  it('can work on JSON with a key named meta', async () => {
    const rawJson = await loadScenario(3);
    expect(parse(rawJson)).toMatchSnapshot();
    expect(getMetaFromPath(rawJson, '/meta/isMeta')).toMatchSnapshot();
  });

  it('can work on JSON with Array', async () => {
    const rawJson = await loadScenario(4);
    expect(parse(rawJson)).toMatchSnapshot();
    expect(getMetaFromPath(rawJson, '/arr/1/foo')).toMatchSnapshot();
  });
});
