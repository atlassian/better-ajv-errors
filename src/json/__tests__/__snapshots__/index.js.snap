// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON can work on JSON with Array 1`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 19,
      "line": 6,
      "offset": 60,
    },
    "source": null,
    "start": Object {
      "column": 14,
      "line": 6,
      "offset": 55,
    },
  },
  "raw": "\\"bar\\"",
  "type": "Literal",
  "value": "bar",
}
`;

exports[`JSON can work on JSON with Array 2`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 12,
      "line": 6,
      "offset": 53,
    },
    "source": null,
    "start": Object {
      "column": 7,
      "line": 6,
      "offset": 48,
    },
  },
  "raw": "\\"foo\\"",
  "type": "Identifier",
  "value": "foo",
}
`;

exports[`JSON can work on JSON with Array with empty children 1`] = `"/arr/4"`;

exports[`JSON can work on JSON with a key named meta 1`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 19,
      "line": 4,
      "offset": 48,
    },
    "source": null,
    "start": Object {
      "column": 15,
      "line": 4,
      "offset": 44,
    },
  },
  "raw": "true",
  "type": "Literal",
  "value": true,
}
`;

exports[`JSON can work on JSON with a key named meta 2`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 13,
      "line": 4,
      "offset": 42,
    },
    "source": null,
    "start": Object {
      "column": 5,
      "line": 4,
      "offset": 34,
    },
  },
  "raw": "\\"isMeta\\"",
  "type": "Identifier",
  "value": "isMeta",
}
`;

exports[`JSON can work on JSON with a key named value 1`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 14,
      "line": 3,
      "offset": 31,
    },
    "source": null,
    "start": Object {
      "column": 12,
      "line": 3,
      "offset": 29,
    },
  },
  "raw": "20",
  "type": "Literal",
  "value": 20,
}
`;

exports[`JSON can work on JSON with a key named value 2`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 10,
      "line": 3,
      "offset": 27,
    },
    "source": null,
    "start": Object {
      "column": 3,
      "line": 3,
      "offset": 20,
    },
  },
  "raw": "\\"value\\"",
  "type": "Identifier",
  "value": "value",
}
`;

exports[`JSON can work on simple JSON 1`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 15,
      "line": 2,
      "offset": 16,
    },
    "source": null,
    "start": Object {
      "column": 10,
      "line": 2,
      "offset": 11,
    },
  },
  "raw": "\\"bar\\"",
  "type": "Literal",
  "value": "bar",
}
`;

exports[`JSON can work on simple JSON 2`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 8,
      "line": 2,
      "offset": 9,
    },
    "source": null,
    "start": Object {
      "column": 3,
      "line": 2,
      "offset": 4,
    },
  },
  "raw": "\\"foo\\"",
  "type": "Identifier",
  "value": "foo",
}
`;

exports[`JSON can work with unescaped JSON pointers with ~0 1`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 17,
      "line": 7,
      "offset": 93,
    },
    "source": null,
    "start": Object {
      "column": 16,
      "line": 7,
      "offset": 92,
    },
  },
  "raw": "2",
  "type": "Literal",
  "value": 2,
}
`;

exports[`JSON can work with unescaped JSON pointers with ~1 1`] = `
Object {
  "loc": Object {
    "end": Object {
      "column": 17,
      "line": 4,
      "offset": 49,
    },
    "source": null,
    "start": Object {
      "column": 16,
      "line": 4,
      "offset": 48,
    },
  },
  "raw": "1",
  "type": "Literal",
  "value": 1,
}
`;

exports[`JSON should not throw error when children is array 1`] = `"/arr/3"`;
